cfme/tests/services/test_provision_stack.py

```diff
@@ -234,21 +244,54 @@ def test_remove_template_provisioning(setup_provider, provider, provisioning, ca
                   dialog=dialog_name_new, orch_template=template.template_name,
                   provider_type=provider.name)
     catalog_item.create()
-    if provider.type == 'ec2':
-        stack_data = {
-            'stack_name': "stack" + fauxfactory.gen_alphanumeric()
-        }
-    elif provider.type == 'openstack':
-        stack_data = {
-            'stack_name': "stack" + fauxfactory.gen_alphanumeric()
-        }
+    stackname = "test-" + fauxfactory.gen_alphanumeric()
+    stack_data = {'stack_name': stackname}
     service_catalogs = ServiceCatalogs("service_name", stack_data)
     service_catalogs.order_stack_item(catalog.name, catalog_item)
     flash.assert_no_errors()
-    template.delete()
+    template.delete_all_templates()
```

Comment:
content of two templates can't be same . so before adding trying to delete template that are not read only and can have same content . Not able to delete individual template as part of clean up code after each test because of a bug https://bugzilla.redhat.com/show_bug.cgi?id=1295941